dataeng/resources/warehouse-transforms-ci-dbt.sh

```diff
@@ -11,4 +11,17 @@ dbt deps --profiles-dir $WORKSPACE/analytics-secure/warehouse-transforms/ --prof
 dbt seed --profiles-dir $WORKSPACE/analytics-secure/warehouse-transforms/ --profile $DBT_PROFILE --target $DBT_TARGET
 
 dbt run $DBT_RUN_OPTIONS $DBT_RUN_EXCLUDE --profiles-dir $WORKSPACE/analytics-secure/warehouse-transforms/ --profile $DBT_PROFILE --target $DBT_TARGET
-dbt test $DBT_TEST_OPTIONS $DBT_TEST_EXCLUDE --profiles-dir $WORKSPACE/analytics-secure/warehouse-transforms/ --profile $DBT_PROFILE --target $DBT_TARGET
\ No newline at end of file
+dbt test $DBT_TEST_OPTIONS $DBT_TEST_EXCLUDE --profiles-dir $WORKSPACE/analytics-secure/warehouse-transforms/ --profile $DBT_PROFILE --target $DBT_TARGET && true
+if [ $? -eq 1 ]
+then
+    if [ "$WITH_RETRY" == "true" ]
+    then
+        pip install -r ../../tools/ci_scripts/requirements.txt
+        NO_OF_TRIES=3
```

Comment:
nit: I feel like it might be better to keep all of the variables together in the `parameters` block in the DSL, so that we don't have to dig in the scripts